src/cc_mm.c

```diff
@@ -82,6 +82,22 @@ _cc_realloc(void *ptr, size_t size, const char *name, int line)
     return p;
 }
 
+void *
+_cc_realloc_test(void *ptr, size_t size, const char *name, int line)
+{
+    void *p = NULL, *pr;
+
+    if ((pr = realloc(ptr, size)) == NULL || (p = malloc(size)) == NULL) {
```

Comment:
I guess the current impl does save you one `free(p)` when allocation fails. So feel free to leave the above rewrite and just add the comment somewhere.